pkg/kubelet/kubelet.go

```diff
@@ -1580,31 +1584,27 @@ func (kl *Kubelet) SyncPods(allPods []api.Pod, podSyncTypes map[types.UID]metric
 
 	// Kill any containers we don't need.
 	killed := []string{}
-	for ix := range dockerContainers {
-		// Don't kill containers that are in the desired pods.
-		dockerName, _, err := dockertools.ParseDockerName(dockerContainers[ix].Names[0])
-		_, found := desiredPods[dockerName.PodUID]
-		if err == nil && found {
+	for _, pod := range runningPods {
```

Comment:
We might misread the logic here, but shouldn't you introduce a bug in a update case described as the below:

 A running pod with two containers: c1 and c2, now update pod with only c1. With your new change, c2 couldn't be removed?